src/modules/sensors/vehicle_imu/VehicleIMU.cpp

```diff
@@ -364,18 +370,24 @@ void VehicleIMU::UpdateIntegratorConfiguration()
 		_accel_integrator.set_reset_interval(roundf((accel_integral_samples - 0.5f) * _accel_interval.update_interval));
 		_gyro_integrator.set_reset_interval(roundf((gyro_integral_samples - 0.5f) * _gyro_interval.update_interval));
 
-		_sensor_accel_sub.set_required_updates(accel_integral_samples);
-		_sensor_gyro_sub.set_required_updates(gyro_integral_samples);
+		// gyro: find largest integer multiple of gyro_integral_samples
+		for (int n = sensor_gyro_s::ORB_QUEUE_LENGTH; n > 0; n--) {
```

Comment:
I guess the tradeoff is that we use more CPU, in particular if gyro_integral_samples is prime and > Qsize.

Reply:
In the "real" cases that motivated the change it was less CPU overall compared to increasing IMU_INTEG_RATE, which then made ekf2 and the attitude controller(s) run more often.

Example
 - icm20689 at 2000 Hz
 - VehicleIMU runing at 400 Hz to gather all data and publish vehicle_imu at 200 Hz

Now I'm wondering if we should care about the possibility of someone accidentally configuring a pathological case. As a start to limit the scope a bit I'll add typical IMU_INTEG_RATE enum values (likely just 100, 200, 400 Hz) to discourage hand tweaking or at least indicate uncharted territory.

Reply:
yeah we can leave it at that. I was thinking of (future) IMU's with exotic update rates.